cmd/tools/test/find_altered_tests.go

```diff
@@ -0,0 +1,163 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// find_altered_tests.go
+//
+// This program identifies altered or added test suite names based on the provided criteria.
+// It accepts the following inputs via command-line flags:
+//
+// - Category (-c): The category of tests to find (e.g., unit, integration).
+// - Source Git Reference (-source-ref): The source Git reference (commit SHA, branch, etc.).
+// - Target Git Reference (-target-ref): The target Git reference (commit SHA, branch, etc.).
+// - Test Directories (-d): Comma-separated list of test directories to search.
+//
+// The program outputs the names of altered test suites, separated by '|'.
+//
+// Usage:
+//   find_altered_tests -c <category> -source-ref <sourceRef> -target-ref <targetRef> -d <testDirs>
+
+func main() {
+	var category string
+	var sourceRef string
+	var targetRef string
+	var testDirs string
+
+	flag.StringVar(&category, "c", "", "Category of altered tests to find")
+	flag.StringVar(&sourceRef, "s", "", "Source Git reference (commit SHA, branch, etc.)")
+	flag.StringVar(&targetRef, "t", "", "Target Git reference (commit SHA, branch, etc.)")
+	flag.StringVar(&testDirs, "d", "", "Comma-separated list of test directories")
+	flag.Parse()
+
+	if category == "" || sourceRef == "" || targetRef == "" || testDirs == "" {
+		fmt.Println("Usage: find_altered_tests -c <category> -s <sourceRef> -t <targetRef> -d <testDirs>")
+		os.Exit(1)
+	}
+
+	modifiedFiles, err := getModifiedTestFiles(sourceRef, targetRef)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting modified test files: %v\n", err)
+		os.Exit(1)
```

Comment:
stdlib log is simpler:
```suggestion
		log.Fatalf("Error getting modified test files: %v", err)
```